File: Latihan a/java/hardware.java
// deklarasi kelas hardware inherit produk
class hardware extends product{
    // variabel privat
    private String brand;
    private String model;
    // method public dan konstruktor
    public hardware(){

    }
    public hardware(String brand, String model){
        this.brand=brand;
        this.model=model;
    }
    // get dans set semua variabel
    public String getbrand(){
        return brand;
    }
    public void setbrand(String brand){
        this.brand = brand;
    }
    public String getmodel(){
        return model;
    }
    public void setmodel(String model){
        this.model = model;
    }
}

File: Latihan b/php/Hardware.php
<?php
// deklarasi kelas hardware inherit produk
class Hardware extends Product{
	//variabel lokal
	private $brand;
	private $model;
	//konstruktor
	public function __construct(){
		$this->brand = "";
        $this->model = "";
	}
	/*
	public function __construct($brand = "", $model = ""){
		$this->brand = $brand;
		$this->model = $model;
	}
	*/
	//get dan set setiap variabel
	public function setbrand($brand){
		$this->brand = $brand;
	}
	public function getbrand(){
		return $this->brand;
	}
	public function setmodel($model){
		$this->model = $model;
	}
	public function getmodel(){
		return $this->model;
	}
	public function __destruct(){
	}
}
?>